src/Mvc/Mvc.Razor/src/RazorPage.cs

```diff
@@ -274,40 +274,11 @@ public override void EnsureRenderedBodyOrSections()
             }
         }
 
-        public override void BeginContext(int position, int length, bool isLiteral)
-        {
-            const string BeginContextEvent = "Microsoft.AspNetCore.Mvc.Razor.BeginInstrumentationContext";
-
-            if (DiagnosticSource?.IsEnabled(BeginContextEvent) == true)
-            {
-                DiagnosticSource.Write(
-                    BeginContextEvent,
-                    new
-                    {
-                        httpContext = Context,
-                        path = Path,
-                        position = position,
-                        length = length,
-                        isLiteral = isLiteral,
-                    });
-            }
-        }
+        public override void BeginContext(int position, int length, bool isLiteral) 
+            => DiagnosticListener.BeginInstrumentationContext(Context, Path, position, length, isLiteral);
```

Comment:
@pranavkm @NTaylorMullen  - we no longer call this from the generated code right?

Reply:
We don't generate code that calls these methods anymore. Can we have these no-op instead?

Reply:
Yup, we don't gen code for this. Honestly, we should at the very least obsolete these methods for 3.0

Reply:
obsolete and make `sealed`; they are called from a couple methods in the code base, so I'll let you do it in another PR. Have nooped them.